BUILD.bazel

```diff
@@ -0,0 +1,291 @@
+    "src/Exception_TEST.cc",
+    "src/Filesystem_TEST.cc",
+    "src/FrameSemantics_TEST.cc",
+    "src/Frame_TEST.cc",
+    "src/Geometry_TEST.cc",
+    "src/Gui_TEST.cc",
+    # "src/ign_TEST.cc",
+    "src/Imu_TEST.cc",
+    "src/JointAxis_TEST.cc",
+    "src/Joint_TEST.cc",
+    "src/Lidar_TEST.cc",
+    "src/Light_TEST.cc",
+    "src/Link_TEST.cc",
+    "src/Magnetometer_TEST.cc",
+    "src/Material_TEST.cc",
+    "src/Mesh_TEST.cc",
+    "src/Model_TEST.cc",
+    "src/Noise_TEST.cc",
+    "src/Param_TEST.cc",
+    "src/parser_TEST.cc",
+    "src/parser_urdf_TEST.cc",
+    "src/Pbr_TEST.cc",
+    "src/Physics_TEST.cc",
+    "src/Plane_TEST.cc",
+    "src/Root_TEST.cc",
+    "src/Scene_TEST.cc",
+    "src/SDF_TEST.cc",
+    "src/SemanticPose_TEST.cc",
+    "src/Sensor_TEST.cc",
+    "src/Sphere_TEST.cc",
+    "src/Surface_TEST.cc",
+    "src/Types_TEST.cc",
+    "src/Utils_TEST.cc",
+    "src/Visual_TEST.cc",
+    "src/World_TEST.cc",
+]
+
+public_headers_no_gen = [
+    "include/sdf/Actor.hh",
+    "include/sdf/AirPressure.hh",
+    "include/sdf/Altimeter.hh",
+    "include/sdf/Assert.hh",
+    "include/sdf/Atmosphere.hh",
+    "include/sdf/Box.hh",
+    "include/sdf/Camera.hh",
+    "include/sdf/Collision.hh",
+    "include/sdf/Console.hh",
+    "include/sdf/Cylinder.hh",
+    "include/sdf/Element.hh",
+    "include/sdf/Error.hh",
+    "include/sdf/Exception.hh",
+    "include/sdf/Filesystem.hh",
+    "include/sdf/Frame.hh",
+    "include/sdf/Geometry.hh",
+    "include/sdf/Gui.hh",
+    "include/sdf/Imu.hh",
+    "include/sdf/JointAxis.hh",
+    "include/sdf/Joint.hh",
+    "include/sdf/Lidar.hh",
+    "include/sdf/Light.hh",
+    "include/sdf/Link.hh",
+    "include/sdf/Magnetometer.hh",
+    "include/sdf/Material.hh",
+    "include/sdf/Mesh.hh",
+    "include/sdf/Model.hh",
+    "include/sdf/Noise.hh",
+    "include/sdf/Param.hh",
+    "include/sdf/parser.hh",
+    "include/sdf/Pbr.hh",
+    "include/sdf/Physics.hh",
+    "include/sdf/Plane.hh",
+    "include/sdf/Root.hh",
+    "include/sdf/Scene.hh",
+    "include/sdf/SDFImpl.hh",
+    "include/sdf/SemanticPose.hh",
+    "include/sdf/Sensor.hh",
+    "include/sdf/Sphere.hh",
+    "include/sdf/Surface.hh",
+    "include/sdf/system_util.hh",
+    "include/sdf/Types.hh",
+    "include/sdf/Visual.hh",
+    "include/sdf/World.hh",
+]
+
+private_headers = [
+    "src/ExceptionPrivate.hh",
+    "src/ign.hh",
+    "src/parser_private.hh",
+    "src/SDFExtension.hh",
+    "src/SDFImplPrivate.hh",
+]
+
+sources = [
+    "src/Actor.cc",
+    "src/AirPressure.cc",
+    "src/Altimeter.cc",
+    "src/Atmosphere.cc",
+    "src/Box.cc",
+    "src/Camera.cc",
+    "src/Collision.cc",
+    "src/Console.cc",
+    "src/Converter.cc",
+    "src/Cylinder.cc",
+    "src/Element.cc",
+    "EmbeddedSdf.cc",
+    "src/Error.cc",
+    "src/Exception.cc",
+    "src/Filesystem.cc",
+    "src/Frame.cc",
+    "src/FrameSemantics.cc",
+    "src/Geometry.cc",
+    "src/Gui.cc",
+    "src/ign.cc",
```

Comment:
remove `ign.cc` from the sources if we are not building a shared library? the reason is `ignitionVersion` symbol is the same across libraries; causing an ODR issue. other instances of `ign.cc` i saw are in `ign_msgs` and `ign_transport`

Reply:
Removed, though this will cause `ign sdf` command line tool to not be available, should I add a separate `cc_binary` for that?